pkg/cache/tas_flavor_snapshot.go

```diff
@@ -356,6 +357,23 @@ func levelKey(topologyRequest *kueue.PodSetTopologyRequest) *string {
 	return nil
 }
 
+// findMostAllocatedDomainIdx finds an index of the first domain with the lowest
+// value of state, higher or equal than count.
+// If such a domain doesn't exist, it returns 0 as it's an index of the domain with the
+// most available resources
+func findMostAllocatedDomainIdx(domains []*domain, count int32) int {
+	mostAllocatedFitIdx := 0
+	for i, domain := range domains {
+		if domain.state < count {
+			break
+		}
+		if domain.state != domains[mostAllocatedFitIdx].state {
+			mostAllocatedFitIdx = i
+		}
```

Comment:
IIUC this check is just to choose the first domain when there is a tie. 

I guess this check chooses the first domain, while without it the last would be chosen in the island of the same "state". 

Choosing the domain in case of a tie should not matter, but choosing the first will affect fewer tests for sure.

Reply:
I believe choosing the first domain among domains with the same state actually simplifies testing and debugging. Imagine you have 4 hosts spread between 2 racks, and you want to schedule 3 pods, then if we didn't choose the first domain we would schedule on `x1,x2` and `x4` hosts. It doesn't impact actual performance of the job, or general fragmentation but I think it's would be easier to reason about consecutive hosts `x1, x2, x3`

Reply:
yeah, makes sense. I'm ok with that

Reply:
I see, I've added the comment